src/components/AppBar.test.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import AppBarPanel from "./AppBar.component";
+
+
+const mockNavigate = jest.fn();
+jest.mock("react-router-dom", () => ({
+  useNavigate: jest.fn().mockImplementation(() => {
+      return mockNavigate;
+  }),
+  useLocation: () => ({
+    pathname: "localhost:3000"
+  }),
+  useParams: () => {
+      return { '*': '' };
+  },
+}));
```

Comment:
For hooks, instead of mocking them, use `spyOn` to mock value for individual tests